docs/langref/relay_expr.rst

```diff
@@ -0,0 +1,396 @@
+====================
+Expressions in Relay
+====================
+
+The Relay IR is a pure, expression-oriented language with distinct
+dataflow and control flow language fragments.
+The dataflow fragments of a program (i.e., those without
+calls to recursive functions or branching) can be
+viewed as a traditional computation graph when writing and
+expressing transformations.
+
+The below sections describe the different expressions in Relay
+as well as give details of their semantics.
+
+Variables
+=========
+
+Relay allows for local and global variables. Our design is based on
```

Comment:
Inspired by LLVM, Relay explicitly distinguishes between local and global variables both in the AST and in the text format. In the text format, global and local variables are distinguished by prefixes, or _sigils_. Global variables are prefixed with `@` and local variables with `%`. This explicit distinction makes certain operations easier. For example inlining a global definition requires no analysis: simply inlining the definition suffices.

Reply:
meta comment: the example is confusing. It's not clear why global and local variables should have different inlining rules.

Reply:
Local vars have scope considerations